nested_lookup/lookup_api.py

```diff
@@ -1,22 +1,24 @@
 import copy
+import warnings
 from six import iteritems
+from nested_lookup import nested_lookup
 
 
-def nested_delete(document, key, in_place=False):
+def nested_delete(document: dict, key: str, in_place: bool = False) -> dict:
```

Comment:
A document may be either a `list` or a `dict`. I'm even thinking about entertaining any iterable.

Additionally because the `document` could be many different types, the return is also not necessarily a `dict`.

All that said the type annotations are new to me, so if they don't support many different types, I think maybe we are better off removing them.